README.md

```diff
@@ -133,10 +138,26 @@ And then inside your components, you can access the context as follows
 ```Typescript
 const { state, signIn, signOut } = useAuthContext();
 ```
+
+Few common methods that you will require when implementing authentication capabilities in your application.
+-   [`signIn`](API.md#signin) - Initiate a login request to Asgardeo, process the response to obtain authentication response.  
+-   [`signOut`](API.md#signout) - Logout the user from Asgardeo and clear any authentication data from the SDK storage.
+-   [`isAuthenticated`](API.md#isauthenticated) - Check whether there is a authenticated user session. Based on the result you can decide to change the application view/behaviour.
+-   [`getBasicUserInfo`](API.md#getbasicuserinfo) - Get authenticated user's basic information from the authencation response.
+-   [`getDecodedIDToken`](API.md#getdecodedidtoken) - Get the decoded id_token obtained in the authentication response. From there you can derive more information such as additinal user-attributes.
+-   [`getIDToken`](API.md#getidtoken) - Get the id_token (JWT) obtained in the authentication response. 
+-   [`getAccessToken`](API.md#getaccesstoken) - Get the access_token obtained in the authentication response. 
+-   [`refreshAccessToken`](API.md#refreshaccesstoken) - Get the refresh_token obtained in the authentication response. 
+-   [`getHttpClient`](API.md#gethttpclient) - Get an HttpClient instance so that the you can make RESTful calls to the backend, where the client will attach the necessory Authorization headers to the request. 
```

Comment:
```suggestion
-   [`getHttpClient`](API.md#gethttpclient) - Get an HttpClient instance so that the you can make RESTful calls to the backend, where the client will attach the necessary Authorization headers to the request. 
```